orbit/pkg/update/flag_runner.go

```diff
@@ -95,6 +95,14 @@ func (r *FlagRunner) DoFlagsUpdate() (bool, error) {
 
 	// next GetConfig from Fleet API
 	flagsJSON, err := r.orbitClient.GetConfig(r.opt.OrbitNodeKey)
+	// on 401 unauthenticated error, re-enroll and update orbit node key
+	if errors.Is(err, service.ErrUnauthenticated) {
+		r.opt.OrbitNodeKey, err = r.updateOrbitNodeKey()
+		if err != nil {
+			return false, err
+		}
+		return false, nil
```

Comment:
Probably ok for now, but DoFlagsUpdate will fail on this call and would have to wait for next call. (This is one of the reasons for having the re-enroll on auth-error logic solved by `r.orbitClient`).

Reply:
Mhm... now am worried that if we have bad flags for osquery (that make it crash) AND the node key has been invalidated then if Orbit restarts it won't do the flags update, and may be loop-crashing? (due to osquery loop crashing)
```go
		// do the initial flags update
		_, err = flagRunner.DoFlagsUpdate()
		if err != nil {
			// just log, OK to continue, since we will retry
			log.Info().Err(err).Msg("Initial flags update failed")
		}
```
But this is probably an edge-case, to solve next release?

Reply:
I just tried this scenario, invalidated orbit node key using `watchdog_delay` flag and giving it a string value..osqueryd crashed as expected with `illegal value` but only once..I corrected the flag, and on the next restart `DoFlagsUpdate()`, gets a 401, updates the node key, and updates the flag...

```
if errors.Is(err, service.ErrUnauthenticated) {
		r.opt.OrbitNodeKey, err = r.updateOrbitNodeKey()
```

^ `DoFlagsUpdate()` re-enrolls on `ErrUnauthenticated` error

Reply:
Ah gotcha, it will work on the second restart (because on the first one it will re-enroll w/ new node key).